### PR TITLE
[Delivers #101710542] Unify naming of client-specific code

### PR DESCRIPTION
**Pull request by jessieay**
* Previously, used `UseCase`, `model_instnace`, and `proposal_delegate`
* Move to variations of `client_data` everywhere
* Story: https://www.pivotaltracker.com/story/show/101710542

### CODE REVIEW

File: app/models/concerns/client_data_type.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
-module ProposalDelegate
+module ClientDataType
```
Comment (jessieay) [inline app/models/concerns/client_data_type.rb]: This is the rename I struggled with most. thoughts?

Comment (pkarman) [inline app/models/concerns/client_data_type.rb]: `ClientDataType` suggests it is an instantiate-able class. How about `ClientDataMixin` or `ClientDataConcern` or `ClientDataExtensions` ?

**Comment by pkarman** [comment]
I agree that `ClientDataType` feels awkward. All the other renames seem sensible.

File: app/models/concerns/client_data_type.rb (hunk 1 above)
Comment (jessieay) [inline app/models/concerns/client_data_type.rb]: I feel like `Mixin` and `Concern` are good at communicating the purpose of this module, but it's also a bit repetitive - like naming a class `WorkOrderClass`. The example I see for ActiveModel Concerns is always `Taggable` -- trying to think of names that are in that spirit

What about just calling it `ClientData` ?

Comment (pkarman) [inline app/models/concerns/client_data_type.rb]: Simply `ClientData` is misleading. It isn't client data. It's the basic association definitions and validation rules that ClientData-type model classes should implement. It's like an abstract class for client data models, but as a concern.

`Mixin` has precedent in `PurchaseCardMixin` and doesn't repeat anything -- there is no `mixin` keyword in Ruby afaik.

Comment (jessieay) [inline app/models/concerns/client_data_type.rb]: Ok, going with`ClientDataMizin

**Pull request merged**